lib/index.ts

```diff
@@ -40,46 +42,63 @@ function buildDepTree(targetFileRaw, lockFileRaw, options) {
     name: targetFile.name || undefined,
     version: targetFile.version || undefined,
   };
-  const parentDepList = targetFile.dependencies;
+
   const fullDepList = lockFile.dependencies;
+  const topLevelDeps = Object.keys(targetFile.dependencies);
+
+  for (const dep of topLevelDeps) {
+    depTree.dependencies[dep] = await buildSubTreeRecursive(dep, []);
+  }
+
+  return depTree;
+
+  async function buildSubTreeRecursive(dep: string, depKeys: string[]) {
 
-  const parentDepsMap = Object.keys(parentDepList).reduce((acc, depName) => {
-    const version = parentDepList[depName];
-    const name = `${depName}@${version}`;
-    acc[name] =  {
-      name: depName,
-      version,
+    const depSubTree = {
+      dependencies: {},
+      name: dep,
+      version: undefined,
     };
-    return acc;
-  }, {});
-
-  const depsMap = Object.keys(fullDepList).reduce((acc, dep) => {
-    const version = fullDepList[dep];
-    const name = `${dep}@${version}`;
-    acc[name] = dep;
-    return acc;
-  }, {});
-
-  for (const dep in depsMap) {
-    if (depsMap.hasOwnProperty(dep)) {
-      const subTree = buildSubTreeRecursive(dep, new Set(), depsMap);
-
-      if (subTree) {
-        depTree.dependencies[subTree.name] = subTree;
+
+    // Get path to the nested dependencies from list ['package1', 'package2']
+    // to ['dependencies', 'package1', 'dependencies', 'package2', 'dependencies']
+    const depPath = getDepPath(depKeys);
+    // try to get list of deps on the path
+    const deps = _.get(lockFile, depPath);
+
+    // If exists and looked-up dep is there
+    if (deps && deps[dep]) {
+      // update the tree
+      depSubTree.version = deps[dep].version;
+      // repeat the process for dependencies of looked-up dep
+      if (deps[dep].requires) {
+        Object.keys(deps[dep].requires).forEach(async (subDep) => {
+          depSubTree.dependencies[subDep] = await buildSubTreeRecursive(subDep, [...depKeys, subDep]);
+
+        });
+        return depSubTree;
+      } else {
+        // no more deps, return tree
+        return depSubTree;
```

Comment:
don't think you need an `else` here since you return in the statement above

Reply:
You are right, I'll move return to the separate statement.